test/unit/source.js

```diff
@@ -30,104 +31,260 @@ global.URL = function URL() {
     this.ref = undefined;
 };
 
-const paramsWMS = {
-    url: 'http://',
-    name: 'name',
-    format: 'image/png',
-    extent: [-90, 90, -45, 45],
-    projection: 'EPSG:4326',
+const vendorSpecific = {
+    buffer: 4096,
+    format_options: 'dpi:300;quantizer:octree',
+    tiled: true,
 };
 
-const paramsWMTS = {
-    url: 'http://',
-    name: 'name',
-    format: 'image/png',
-};
-
-const paramsTMS = {
-    url: 'http://',
-};
-
-describe('Source', function () {
-    it('Should instance and throw error for Source', function () {
-        const source = new Source({ url: 'http://'  });
-        assert.throws(source.urlFromExtent, Error);
-        assert.throws(source.extentInsideLimit, Error);
-        assert.throws(source.extentsInsideLimit, Error);
+describe('Sources', function () {
+    beforeEach(function () {
+        sinon.spy(console, 'warn');
```

Comment:
The test fails if `sinon` finds a warn console?

Reply:
Before each test, `sinon` is spying on what we tell it: here it is `console.warn`. It modifies the `console.warn` method, and allow us to do some tests later on (see below `console.warn.calledOnce` for example).